core/controllers/acl_decorators.py

```diff
@@ -2615,3 +2617,153 @@ def test_can_play_entity(self, entity_type, entity_id, **kwargs):
+        if (
+                role_services.ACTION_ACCEPT_ANY_VOICEOVER_APPLICATION not in
+                user_actions_info.actions):
+            raise self.UnauthorizedUserException(
+                'You do not have credentials to review voiceover '
+                'applications.')
+
+        return handler(self, voiceover_application_id, **kwargs)
+
+    test_can_review_voiceover_application.__wrapped__ = True
+
+    return test_can_review_voiceover_application
+
+
+def can_submit_voiceover_application(handler):
+    """Decorator to check whether a user can submit voiceover application.
+
+    Args:
+        handler: function. The function to be decorated.
+
+    Returns:
+        function. The newly decorated function that now also checks
+            if the user has permission to submit voiceover application.
+    """
+
+    def test_can_submit_voiceover_application(self, **kwargs):
+        """Checks if the user can submit voiceover application.
+
+        Args:
+            **kwargs: *. Keyword arguments.
+
+        Returns:
+            *. The return value of the decorated function.
+
+        Raises:
+            UnauthorizedUserException: The user does not have
+                credentials to submit voiceover application.
+        """
+        if role_services.ACTION_SUBMIT_VOICEOVER_APPLICATION in (
+                self.user.actions):
+            # The voiceover_service returns user submitted voiceover
+            # applications in sorted order of the date of submission.
+            voiceover_applications = (
+                voiceover_services.get_user_submitted_voiceover_applications(
+                    self.user_id))
+            if voiceover_applications:
+                latest_voiceover_application = voiceover_applications[0]
+                if latest_voiceover_application.status == (
+                        suggestion_models.STATUS_IN_REVIEW):
+                    raise self.UnauthorizedUserException(
+                        'You can only submit one application at a time. '
+                        'It looks like your latest application is not yet '
+                        'reviewed.')
+                elif latest_voiceover_application.status == (
+                        suggestion_models.STATUS_ACCEPTED):
+                    if latest_voiceover_application.target_type == (
+                            suggestion_models.TARGET_TYPE_EXPLORATION):
+                        exploration = exp_fetchers.get_exploration_by_id(
+                            latest_voiceover_application.target_id)
+                        if exploration.has_complete_voiceover(
+                                latest_voiceover_application.language_code):
+                            return handler(self, **kwargs)
+                        else:
+                            raise self.UnauthorizedUserException(
+                                'You cannot submit a new application until the '
```

Comment:
So what if it is completed, and then you submit an application, but then the exploration has some new states, which aren't completed yet?

I think this should be more of a setting on the creator side than on us devs right? The creators should be able to decide if their applicant can work multiple at a time or not, and we should display this info upfront somewhere.

Reply:
> So what if it is completed, and then you submit an application, but then the exploration has some new states, which aren't completed yet?

We are not considering the older exploration which you are assigned to, we will be showing voiceover status on the "My contributions" tab of the dashboard page and if the user see the needs then they will do the voiceover accordingly. (Note: Contributors can do voiceover to new content (or updated content) only if there exists a translation for the content.)

Reply:
Does this block the user from taking on new exps in the current coded checks?